src/order/sup_indep.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.lattice
+import data.set.pairwise
+
+/-!
+# Finite supremum independence
+
+In this file, we define supremum independence of indexed sets. An indexed family `f : ι → α` is
+sup-independent if, for all `a`, `f a` and the supremum of the rest are disjoint.
+
+In distributive lattices, this is equivalent to being pairwise disjoint.
+
+## TODO
+
+As we don't have `lattice_bot`, we're forced to pick between `bounded_lattice` (which assumes `top`
+unnecessarily) and `distrib_lattice_bot` (which assumes distributivity unnecessarily). For now we
+pick `distrib_lattice_bot` for it to apply to `finset α`, although this use is redundant with
+`set.pairwise_disjoint`.
```

Comment:
This is no longer true, right? you can use `[order_bot α]` `[lattice α]`?

Reply:
It is, because we don't have (nor want) the `lattice α` + `order_bot α` to `semilattice_inf_bot α` instance.